src/test/java/seedu/address/logic/commands/internships/task/EditInternshipTaskCommandTest.java

```diff
@@ -0,0 +1,160 @@
+package seedu.address.logic.commands.internships.task;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static seedu.address.logic.commands.CommandAssignmentTestUtil.assertCommandSuccess;
+
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.CommandAssignmentTestUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.commands.internship.task.EditInternshipTaskCommand;
+import seedu.address.logic.commands.internship.task.FindInternshipTaskCommand;
+import seedu.address.model.AddressBook;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.fields.Outcome;
+import seedu.address.model.fields.TaskOutcome;
+import seedu.address.model.internship.role.InternshipRole;
+import seedu.address.model.internship.task.InternshipTask;
+import seedu.address.model.internship.task.InternshipTaskNameContainsKeywordsPredicate;
+import seedu.address.testutil.InternshipRoleBuilder;
+import seedu.address.testutil.InternshipTaskBuilder;
+
+public class EditInternshipTaskCommandTest {
+    private final InternshipRole role1 = new InternshipRoleBuilder().withName("Company1")
+            .withRole("SWE").withCycle("Summer").build();
+
+    private final InternshipRole role2 = new InternshipRoleBuilder().withName("Company2")
+            .withRole("SWE").withCycle("Winter").build();
+
+    private final InternshipTask task1BelongingToRole1 = new InternshipTaskBuilder()
+            .withInternshipRole(role1).withTaskName("Interview Preparation_1").build();
+
+    private final InternshipTask task2BelongingToRole1 = new InternshipTaskBuilder()
+            .withInternshipRole(role1).withTaskName("OA_1").build();
+
+    private final InternshipTask task1BelongingToRole2 = new InternshipTaskBuilder()
+            .withInternshipRole(role2).withTaskName("Interview Preparation_2").build();
+
+    private final InternshipTask task2BelongingToRole2 = new InternshipTaskBuilder()
+            .withInternshipRole(role2).withTaskName("OA_2").build();
+
+    private ModelManager model;
+
+    private AddressBook getAddressBook() {
+        AddressBook ab = new AddressBook();
+        ab.addInternshipRole(role1);
+        ab.addInternshipRole(role2);
+        ab.addInternshipTask(task1BelongingToRole1);
+        ab.addInternshipTask(task2BelongingToRole1);
+        ab.addInternshipTask(task1BelongingToRole2);
+        ab.addInternshipTask(task2BelongingToRole2);
+        return ab;
+    }
+
+    private void editTask(InternshipTask taskToBeEdited, InternshipTask editedTask, ModelManager model) {
+        model.setInternshipTask(taskToBeEdited, editedTask);
+    }
```

Comment:
I'm don't quite understand why this is necessary, can't you just call expectedModel.setInternshipTask() within the test?